packages/components/src/tree-grid/index.js

```diff
@@ -149,7 +149,13 @@ function TreeGrid(
 					return;
 				}
 
-				// Focus the next element.
+				// Focus the next column only if expanded.
+				if (
+					nextIndex !== 0 &&
+					activeRow.getAttribute( 'aria-expanded' ) === 'false'
+				) {
+					nextIndex -= 1;
```

Comment:
I think the part that's confusing about this code is that it doesn't take into account whether the user presses right or left keys. I guess the intention is to prevent any changes to the index? Basically stopping focus from moving? I don't know that it's necessary because the `return` statement on line 135 prevents the same thing.

It seems to work better for me if I change it to something like this to prevent focus moving beyond the right edge of the grid:
```js
if (
	keyCode === RIGHT &&
	nextIndex === currentColumnIndex &&
	activeRow.getAttribute( 'aria-expanded' ) === 'false'
) {
	return;
}
```